### PR TITLE
[Merged by Bors] - Cleanup some outdated parts of ecs_guide

### PR DESCRIPTION
**Pull request by DJMcNab**
# Objective

- `Local`s can no longer be accessed outside of their creating system, but these docs say they can be.
- There's also little reason to have a pure wrapper type for `Local`s; they can just use the real type. The parameter name should be sufficiently documenting.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: examples/ecs/ecs_guide.rs
Comment (DJMcNab) [inline examples/ecs/ecs_guide.rs:235]: Ideally this would use the correct suffix, but that should probably be part of an external crate. We don't currently depend on (MPL 2.0)[`ordinal`](https://docs.rs/ordinal/latest/ordinal/) or an equivalent, as far as I know.

**Comment by alice-i-cecile** [comment]
I'm ready to merge once CI passes and I have a second approval :)

**Comment by DJMcNab** [comment]
CI failure appears to be spurious

**Comment by alice-i-cecile** [comment]
bors try

**Comment by DJMcNab** [comment]
bors try

**Comment by alice-i-cecile** [comment]
@bevyengine/docs-team This is ready to go but I need another review.

**Review by IceSentry — APPROVED** [review]

**Pull request closed without merging**
